core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientServiceHandler.java

```diff
@@ -186,18 +202,38 @@ public void completeFile(CompleteFilePRequest request,
   public void createDirectory(CreateDirectoryPRequest request,
       StreamObserver<CreateDirectoryPResponse> responseObserver) {
     CreateDirectoryPOptions options = request.getOptions();
+
     RpcUtils.call(LOG, () -> {
+      if (request.getOptions().getCommonOptions().getCheckS3BucketPath()) {
+        checkBucketPathExists(request.getPath());
+      }
       AlluxioURI pathUri = getAlluxioURI(request.getPath());
       mFileSystemMaster.createDirectory(pathUri, CreateDirectoryContext.create(options.toBuilder())
           .withTracker(new GrpcCallTracker(responseObserver)));
       return CreateDirectoryPResponse.newBuilder().build();
     }, "CreateDirectory", "request=%s", responseObserver, request);
   }
 
+  private void checkBucketPathExists(String path)
+      throws AlluxioException, IOException {
+
+    String bucketPath =
+        AlluxioURI.SEPARATOR + path.split(AlluxioURI.SEPARATOR, 3)[1];
```

Comment:
The hard-coded numbers could cause difficulty to understand and when it is changed later it would be hard to find it here to change. Would it be a better idea to move it to a dedicated place like PathUtils?